enhancements/network/dpu/ocp-on-dpu.md

```diff
@@ -0,0 +1,260 @@
+
+- Add the BF-2 workers to the tenant cluster. The combination of the
+  CNO config in the tenant cluster and the DPU Operator in the infra
+  cluster will cause each BF-2 worker to be configured to run
+  ovn-kubernetes in "DPU Host" mode with its corresponding NIC running
+  ovn-kubernetes in "DPU NIC" mode.
+
+[DPU Operator]: https://github.com/openshift/enhancements/pull/890
+
+#### Upgrade
+
+When upgrading to a new OCP release, each BF-2 node in the infra
+cluster will need to reboot at some point. This will cause a temporary
+(~ 1 minute) network outage on the corresponding tenant cluster host.
+
+In the GA product, we will need to coordinate the upgrades between the
+two clusters, so that each NIC reboots just before its corresponding
+host, after the host has already been drained of pods. However, this
+functionality will not be implemented in the Dev Preview; upgrades
+between the two clusters will be uncoordinated, resulting in
+networking outages.
+
+#### OCP Modifications in the Tenant Cluster
+
+##### ovn-kubernetes (DPU Host mode)
+
+The modifications to ovn-kubernetes itself to support "DPU Host" mode
+have already been made upstream.
+
+##### cluster-network-operator
+
+CNO will need to know when to configure ovn-kubernetes in "DPU Host"
+mode. For Dev Preview, we are not yet ready to add proper
+`Network.operator.openshift.io` API for this, so we will instead have
+CNO react to an optional ConfigMap.
+
+##### Special Handling of BF-2 Host Nodes
+
+When running in "DPU Host" mode, ovn-kubernetes will configure all
+pod-network pods on a node to have SR-IOV interfaces rather than veths
+as their primary network interfaces. Because of how SR-IOV currently
+works in Kubernetes, this requires setting certain fields in the
+`spec` of any pods that will be scheduled on those nodes. This means
+that "ordinary" SR-IOV-unaware pod-network pods cannot be scheduled to
+BF-2 host nodes.
+
+While this will eventually need to be fixed, this means that for Dev
+Preview, we need to ensure that OCP does not attempt to run any of its
+own pod-network pods on BF-2 host nodes. (eg, `dns-default` pods
+cannot run on these nodes). This may involve some combination of
+special configuration, and tainting of the nodes in the BF-2 host
+MachineConfigPool.
+
+#### OCP Modifications in the Infra Cluster
+
+##### ovn-kubernetes (DPU NIC mode)
+
+As with the tenant side, the changes to ovn-kubernetes itself to
+support "DPU NIC" mode have already been made upstream.
+
+Infra cluster masters and non-BF-2 worker nodes will run
+ovn-kubernetes (in "normal" mode) as their CNI plugin. However, BF-2
+NIC nodes will not be able to do this, since it is not currently
+possible to run ovn-kubernetes in both "normal" mode and "DPU NIC"
+mode on the same host.
+
+The current plan for Dev Preview is that BF-2 worker nodes will not
+run any CNI plugin, and will only support host-network pods.
+(Actually, they need _some_ CNI in order for the node to become Ready
```

Comment:
_some_ CNI, not just for node to become Ready, but also for cluster pods (such as MCD, DNS) to function properly.  It is worth mentioning the use of kube-proxy in infra cluster as well.

Reply:
machine-config-daemon is hostNetwork. I thought we had decided we'd just not run CoreDNS on either the DPU Host workers or the DPU NIC workers for Dev Preview?

Reply:
Right about DNS.  ok I thought when we say CNI, it would cover the k8s service networking like accessing from hostnetwork pod to api server, thus the kube-proxy, but if the CNI here is just about pod networking, it shall be fine to mention kube-proxy.